.github/workflows/integration-tests.yml

```diff
@@ -0,0 +1,404 @@
+name: Unit and Integration Tests
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+      - release/v*
+      - refs/tags/v*
+    tags:
+      - v*
+  pull_request:
```

Comment:
Given that this runs on PRs, we must:
1. Do **not** have credentials and tokens exposed in such workflows.
2. Drop all permissions in the workflow for security reasons, or explicitly set only the needed ones. IMO it seems that we can safely drop all permissions with `permissions: {}`.

CCing @rancher/rancher-security to have this in mind when reviewing future GHA PRs.

Reply:
Added it to disable permissions